op-chain-ops/genesis/helpers.go

```diff
@@ -107,3 +108,13 @@ func newHexBig(in uint64) *hexutil.Big {
 	hb := hexutil.Big(*b)
 	return &hb
 }
+
+// CreateAccountOnSetCode is a vm.StateDB wrapper that optimistically creates an account on SetCode
+type CreateAccountOnSetCode struct {
+	vm.StateDB
+}
+
+func (s *CreateAccountOnSetCode) SetCode(addr common.Address, code []byte) {
+	s.CreateAccount(addr)
+	s.StateDB.SetCode(addr, code)
+}
```

Comment:
we can remove this workaround if we add a `db.CreateAccount` call in between these two lines: https://github.com/ethereum-optimism/op-geth/blob/ed8e9f54ed0a5069e77ea34895ff12a9ac4ca17f/consensus/misc/create2deployer.go#L35-L36

turns out that the explicit `CreateAccount` call is required for geth's `MemoryStateDB` but not for the standard `StateDB`, which does something similar to this code

Reply:
We should check to make sure that the code size is zero before calling `CreateAccount` because it will reset the account if it exists. When implementing this originally, i tried to match the `StateDB` behavior 1:1. I am curious if I just implemented it incorrectly or if the behavior of `StateDB` changed

Reply:
At this point I'm erring towards modifying the `MemoryStateDB` to better match `StateDB` rather than adding the `CreateAccount` call upstream, see comment below.